pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go

```diff
@@ -58,7 +65,9 @@ func NewNamespaceController(kubeClient kube.Client, caBundleWatcher *keycertbund
 		caBundleWatcher:           caBundleWatcher,
 		DiscoveryNamespacesFilter: discoveryNamespacesFilter,
 	}
-	c.queue = controllers.NewQueue("namespace controller", controllers.WithReconciler(c.insertDataForNamespace))
```

Comment:
This change is brought after controller queue adopted。

Before that we have no max retry limit, I am good with both a max = 5 or unlimited though. In most cases they have no diff.
@howardjohn WDYT?

Reply:
Maybe it is better to have a default configuration of 5 retries, which has two benefits:

- You can retry failed case.
- A maximum of 5 tries are allowed to prevent the queue from staying in retry state.

Reply:
long time ago. 1.11 was the change. Before it was infinite like you mentioned.

I prefer no infinite, as there are some users that get into a state where 2 istiod's are fighting to reconcile. With infinite retry they may overload API server.